Author: Matus Kral <matuskral@me.com>
Date:   Sun Dec 19 07:08:59 2014 +0100

    Function CecSourceActivated is supposed to pause
playback if another device is activated on CEC bus. this action is triggered
later in the code via app messenger call MediaPause(). MediaPause() before
unpausing video is wakingupfrom screensaver too. this sends 'OnScreensaverDeactivated'
msg what returning back to CecAdapter calls (based on wakeonscreensaver) 
ActivateSource(). 
    By use of player->IsPlaying check (what doesn't consider already paused playback)
whole process can actually make switching to another source impossible - as if playback
is already paused and XBMC gets deactivated, player->Pause() is still called, event
screensaverdeactivated generated and XBMC finally activates itself again. 

diff --git a/xbmc/peripherals/devices/PeripheralCecAdapter.cpp b/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
index 8919380..dd74772 100644
--- a/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
+++ b/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
@@ -1158,7 +1165,7 @@ void CPeripheralCecAdapter::CecSourceActivated(void *cbParam, const CEC::cec_log
     bool bShowingSlideshow = (g_windowManager.GetActiveWindow() == WINDOW_SLIDESHOW);
     CGUIWindowSlideShow *pSlideShow = bShowingSlideshow ? (CGUIWindowSlideShow *)g_windowManager.GetWindow(WINDOW_SLIDESHOW) : NULL;
     bool bPlayingAndDeactivated = activated == 0 && (
-        (pSlideShow && pSlideShow->IsPlaying()) || g_application.m_pPlayer->IsPlaying());
+        (pSlideShow && pSlideShow->IsPlaying()) || !g_application.m_pPlayer->IsPausedPlayback());
     bool bPausedAndActivated = activated == 1 && adapter->m_bPlaybackPaused && (
         (pSlideShow && pSlideShow->IsPaused()) || g_application.m_pPlayer->IsPausedPlayback());
     if (bPlayingAndDeactivated)
